# HG changeset 9712 patch
# User kaf24@firebug.cl.cam.ac.uk
# Node ID 1198f4eb43d10af04adbaaab52bfa55b6f48e9b6
# Parent  ef1a49d180aca500c85969755f80235ccde6eb37
Subject: A few put_cpu() calls were missed when adding CONFIG_X86_NO_TSS.
From: Jan Beulich <jbeulich@novell.com>
xen-unstable changeset:   10261:90a8ab269afe539f87ee4e50607a88ccfbccf3fe
xen-unstable date:        Fri Jun  2 13:36:29 2006 +0100

Index: head-2006-06-06/arch/i386/kernel/vm86.c
===================================================================
--- head-2006-06-06.orig/arch/i386/kernel/vm86.c	2006-06-06 12:51:56.000000000 +0200
+++ head-2006-06-06/arch/i386/kernel/vm86.c	2006-06-06 12:52:27.000000000 +0200
@@ -132,7 +132,9 @@ struct pt_regs * fastcall save_v86_state
 	current->thread.sysenter_cs = __KERNEL_CS;
 	load_esp0(tss, &current->thread);
 	current->thread.saved_esp0 = 0;
+#ifndef CONFIG_X86_NO_TSS
 	put_cpu();
+#endif
 
 	loadsegment(fs, current->thread.saved_fs);
 	loadsegment(gs, current->thread.saved_gs);
@@ -310,7 +312,9 @@ static void do_sys_vm86(struct kernel_vm
 	if (cpu_has_sep)
 		tsk->thread.sysenter_cs = 0;
 	load_esp0(tss, &tsk->thread);
+#ifndef CONFIG_X86_NO_TSS
 	put_cpu();
+#endif
 
 	tsk->thread.screen_bitmap = info->screen_bitmap;
 	if (info->flags & VM86_SCREEN_BITMAP)
